spython/image/__init__.py

```diff
@@ -8,7 +8,30 @@
 import hashlib
 import os
 import re
+from spython.logger import bot
 
+def get_uri(image):
+    '''get the uri of an image, or the string (optional) that appears before
+        ://. Optional. If none found, returns ''
+    '''
+    image = image or ''
+    uri = ''
+
+    match = re.match("^(?P<uri>.+)://", image)
+    if match:
+        uri = match.group('uri')
+
+    return uri
+
+
+def remove_uri(image):
```

Comment:
Why are you removing this from the image client? It means that when a user is interacting with an image object, they cannot get the uri.

Reply:
`get_uri` and `remove_uri` are actually pure functions and unrelated to the image they are attached too

`Instance` and `OciImage` (subclasses of ImageBase) overwrite `get_uri` with a different signature (no argument besides `self`) which is wrong (doesn't work as expected)

And finally: those 2 functions do not get/remove the "URI" but the "protocol" (which is very confusing itself) but the overwritten `get_uri` return the full URI (`foo://bar`)

Reply:
So why make the changes here, but not for the instance class?

Reply:
Those functions are actually unrelated besides having the same name and do different things. I suppose the others are doing what they are supposed to do (they were used before and no one complained) so no need to change them. I'd rather change the names of those if I understood why e.g. `-` or `/` are stripped in a function called `remove_uri`

I'd probably combine both into 1: `splitUri` which returns a `(protocol, name)` pair and use this in `utils.terminal.remove_uri` (another function with the same name and different implementation)

Reply:
The original idea was to have a common function that returns (different) names depending on the kind of class. But you are correct that the utils version came from the previous Python section of the Singularity software (to be consistent) and they are very different.

I'd like you to do what you think is best, your changes and suggestions have been very good so far. I also like the idea of adding pylint for reasonable checks - it's something I've been wanting to do for a while (but was worried it would make contribution harder for others). But after using GoLang a bit, I'm really thinking it should be standard to do checks for unused variables and imports, so I'm leaning on adding it. 

I'm going out for a few hours (need to catch a train) but I'll be back later this evening, so please take your time in your implementation! If you also decide to take a few days to do this, that's okay with me too :) Happy Memorial Day, if that means anything for you, and chat later! :)

Reply:
Alright, will do tomorrow. Have a look at the other checks so far (especially the disabled ones in .pylintrc, all of them cause more warnings...)
See you later.

Reply:
Will do! After we finish this up I’ll open up another PR to go through those in detail. Okay now chat you later for real :)